### ISSUE TITLE
Can I convert a remote docx (on s3) to PDF?

### ISSUE THREAD

**Issue by shaharsol** [yumauri/gotenberg-js-client#14]
Not sure what the syntax would be.

I try 

```
    const signedUrl = await this.getSignedURL(bucket, docxKey);
    console.log('signed url: ', signedUrl);
    const toPDF = pipe(
      gotenberg('http://localhost:8008'),
      convert,
      url,
      office,
      set(filename('result.pdf')),
      please
    )

    const pdfStream = await toPDF(signedUrl);
```

and I always get 404 from Gotenberg as if it wasn't able to fetch the signed url tough it is accessible from curl or browser.

**Comment by yumauri** [comment]
I think, [Gotenberg](https://thecodingmachine.github.io/gotenberg/) doesn't support this, unfortunately :( You can ask for feature [here](https://github.com/thecodingmachine/gotenberg/issues), but for now, as I see, remote URL is considered HTML.

> URL conversions work the same as HTML conversions.

For now, I guess, the only way is to download document and send it to Gotenberg as a file.

**Comment by yumauri** [comment]
(You don't have to save document to the filesystem, if you don't want to — you can use Buffer or ReadableStream)

**Comment by yumauri** [comment]
BTW to clarify, why you get 404:

Functions like `convert`, `url`, `office`, aside from setting conversion type, appends URL parts to the base Gotenberg URL, so in case of

```javascript
const toPDF = pipe(
  gotenberg('http://localhost:8008'),
  convert,
  url,
  office,
  set(filename('result.pdf')),
  please
)
```

you get URL

```
http://localhost:8008/convert/url/office
```

and this is not a valid Gotenberg endpoint, of course.

Not sure, should I handle such cases in my client somehow, or not... I'll think on it

**Comment by yumauri** [comment]
I've added [few lines to README file](https://github.com/yumauri/gotenberg-js-client#office--gotenberg-documentation), with example of streaming downloaded file to Gotenberg, without saving it to the disk.

```javascript
// you can create this function as a static class field or module local variable,
// there is no need to create it anew for each conversion
const toPDF = pipe(
  gotenberg('http://localhost:8008'),
  convert,
  office,
  please
)

// --- 8< ---

const signedUrl = await this.getSignedURL(bucket, docxKey)

// I assume s3 uses HTTPS:
// import https from 'https'
https.get(signedUrl, async (document) => {
  const pdfStream = await toPDF({ 'document.docx': document })
  // ...
})
```

I'll leave this issue open for a while, though, because I want to think about handling issues, when you specify multiple conversion types (which is incorrect and should throw an exception).

**Comment by yumauri** [comment]
Published new version [0.6.1](https://github.com/yumauri/gotenberg-js-client/releases/tag/v0.6.1)

It will throw error

> Cannot set "Office" conversion, already set to "Url"

For your [op code](https://github.com/yumauri/gotenberg-js-client/issues/14#issue-624796661)
